cl/recap/tasks.py

```diff
@@ -1795,11 +1796,46 @@ def get_attachment_page_by_url(att_page_url: str, court_id: str) -> str | None:
     return att_response.text
 
 
+def set_rd_sealed_status(
+    rd: RECAPDocument, magic_number: str | None, potentially_sealed: bool
+) -> None:
+    """Set RD is_sealed status according to the following conditions:
+
+    If potentially_sealed is false, set as not sealed.
+    If potentially_sealed is True and there magic_number, set as sealed.
+    If potentially_sealed is True and no magic_number available, check on PACER
+    if the document is sealed.
+
+    :param rd: The RECAPDocument to set is sealed status.
+    :param magic_number: The magic number if available.
+    :param potentially_sealed: Weather the RD might be sealed or not.
+    :return: None
+    """
+
+    rd.refresh_from_db()
+    if not potentially_sealed:
+        rd.is_sealed = False
+        rd.save()
+        return
+
+    if magic_number:
+        rd.is_sealed = True
+    else:
+        if is_pacer_doc_sealed(
+            rd.docket_entry.docket.court.pk, rd.pacer_doc_id
+        ):
+            rd.is_sealed = True
+        else:
+            rd.is_sealed = False
```

Comment:
This is less code, but I'm not sure it's more clear....hmm.

```suggestion
    rd.is_sealed = True
    if not magic_number and not is_pacer_doc_sealed(
        rd.docket_entry.docket.court.pk, rd.pacer_doc_id
    ):
        rd.is_sealed = False
```

Reply:
Your call if you like this better!

Reply:
Yeah, this is simpler and it's still clear. Changed it.